src/main/java/com/mongodb/kafka/connect/sink/processor/field/projection/FieldProjector.java

```diff
@@ -42,56 +40,83 @@
   static final String SUB_FIELD_DOT_SEPARATOR = ".";
 
   private final Set<String> fields;
+  private final FieldProjectionType fieldProjectionType;
+  private final SinkDocumentField sinkDocumentField;
+
+  public enum SinkDocumentField {
+    KEY,
+    VALUE
+  }
 
-  public FieldProjector(final MongoSinkTopicConfig config, final Set<String> fields) {
+  public FieldProjector(
+      final MongoSinkTopicConfig config,
+      final Set<String> fields,
+      final FieldProjectionType fieldProjectionType,
+      final SinkDocumentField sinkDocumentField) {
     super(config);
     this.fields = fields;
+    this.fieldProjectionType = fieldProjectionType;
+    this.sinkDocumentField = sinkDocumentField;
   }
 
   public Set<String> getFields() {
     return fields;
   }
 
-  protected abstract void doProjection(String field, BsonDocument doc);
-
-  protected static Set<String> getKeyFields(final AbstractConfig config) {
-    return buildProjectionList(
-        config.getString(KEY_PROJECTION_TYPE_CONFIG), config.getString(KEY_PROJECTION_LIST_CONFIG));
+  @Override
+  public void process(final SinkDocument doc, final SinkRecord orig) {
```

Comment:
I moved and simplified the processing logic.  Note we only have ALLOWLIST and BLOCKLIST because the deprecated Blacklist/Whitelist classes now extend the AllowList/BlockList classes.